test/e2e/util/util.go

```diff
@@ -786,6 +787,7 @@ func EnsureNetworkPolicies(t *testing.T, ctx context.Context, c crclient.Client,
 				"control-plane-operator",
 				"hosted-cluster-config-operator",
 				"cloud-controller-manager",
+				"olm-collect-profiles",
```

Comment:
This list is passed to checkPodsHaveLabel.
checkPodsHaveLabel checks this list matches one to one with pods. Atm the test fails when the pod is running because checkPodsHaveLabel finds a pod with the label which is not in the expected list, with this change the test would fail when the pod is not running.
So to make it pass consistently we could modify [this loop](https://github.com/openshift/hypershift/blob/5f604fb0e86fa6491d51ddc12b22a8dbf8c57028/test/e2e/util/util.go#L850) to something like
 
```
		if pod.Name == "olm-collect-profiles" {
			components = append(components, "olm-collect-profiles")
		}
```
or
```
		if pod.Name == "olm-collect-profiles" {
			continue
		}
```